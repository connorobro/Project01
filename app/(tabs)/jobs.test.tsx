/// <reference types="jest" />
import { render } from "@testing-library/react-native";
import JobsScreen from "./jobs";

describe("<JobsScreen />", () => {
  test("shows header and Saved Jobs button", () => {
    const { getByText } = render(<JobsScreen />);
    getByText("Job Listings");
    getByText("Saved Jobs");
  });

  test("renders two placeholder job cards", () => {
    const { getAllByText } = render(<JobsScreen />);
    expect(getAllByText("Job Title").length).toBe(2);
  });

  test("snapshot renders", () => {
    const tree = render(<JobsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
