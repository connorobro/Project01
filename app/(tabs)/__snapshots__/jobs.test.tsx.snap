// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobsScreen /> snapshot renders 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#f4f6f8",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
            "fontWeight": "800",
          }
        }
      >
        Job Listings
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#0ea5a4",
            "borderRadius": 12,
            "paddingHorizontal": 12,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontWeight": "700",
            }
          }
        >
          Saved Jobs
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#d1fae5",
          "borderRadius": 12,
          "borderWidth": 1,
          "height": 42,
          "justifyContent": "center",
          "marginBottom": 12,
          "marginTop": 10,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#6b7280",
          }
        }
      >
        Search (placeholder)
      </Text>
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingBottom": 28,
        }
      }
    >
      <View>
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#d1fae5",
              "borderLeftColor": "#2dd4bf",
              "borderLeftWidth": 4,
              "borderRadius": 16,
              "borderWidth": 1,
              "marginBottom": 12,
              "padding": 14,
            }
          }
        >
          <Text
            style={
              {
                "color": "#0f172a",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            Job Title
          </Text>
          <Text
            style={
              {
                "color": "#6b7280",
                "marginTop": 2,
              }
            }
          >
            Company • Location
          </Text>
          <Text
            style={
              {
                "color": "#1f2937",
                "marginTop": 8,
              }
            }
          >
            Short summary placeholder to show how the card will look.
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 12,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#0ea5a4",
                  "borderRadius": 10,
                  "paddingHorizontal": 12,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffffff",
                    "fontWeight": "700",
                  }
                }
              >
                Save Job
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#6b7280",
                  "textAlign": "right",
                  "width": 96,
                }
              }
            >
              Posted: —
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderColor": "#d1fae5",
              "borderLeftColor": "#2dd4bf",
              "borderLeftWidth": 4,
              "borderRadius": 16,
              "borderWidth": 1,
              "marginBottom": 12,
              "padding": 14,
            }
          }
        >
          <Text
            style={
              {
                "color": "#0f172a",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            Job Title
          </Text>
          <Text
            style={
              {
                "color": "#6b7280",
                "marginTop": 2,
              }
            }
          >
            Company • Location
          </Text>
          <Text
            style={
              {
                "color": "#1f2937",
                "marginTop": 8,
              }
            }
          >
            Short summary placeholder to show how the card will look.
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 12,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#0ea5a4",
                  "borderRadius": 10,
                  "paddingHorizontal": 12,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#ffffff",
                    "fontWeight": "700",
                  }
                }
              >
                Save Job
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#6b7280",
                  "textAlign": "right",
                  "width": 96,
                }
              }
            >
              Posted: —
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 12,
          "justifyContent": "center",
          "marginTop": "auto",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#0ea5a4",
              "borderRadius": 10,
              "paddingHorizontal": 12,
              "paddingVertical": 8,
            },
            {
              "backgroundColor": "#a7f3d0",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontWeight": "700",
            }
          }
        >
          ◀ Prev
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#0f172a",
            "fontWeight": "700",
          }
        }
      >
        Page 1
      </Text>
      <View
        style={
          {
            "backgroundColor": "#0ea5a4",
            "borderRadius": 10,
            "paddingHorizontal": 12,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontWeight": "700",
            }
          }
        >
          Next ▶
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
